### ISSUE TITLE
Announcement: New Feature: InstagramID Shortcode Library

### ISSUE THREAD

**Issue by SteveJobzniak** [mgp25/Instagram-API#1079]
Placing this announcement here for a while for anyone who has ever needed this feature.

https://github.com/mgp25/Instagram-API/commit/0c2b44a591098cd3b3afaff0400604b674abf440

This new library lets you convert Instagram shortcodes to and from internal IDs. The shortcode is the https://instagram.com/p/SHORTCODE/ part of the URL.

There are many reasons why you would want to be able to convert back and forth between shortcodes and internal ID numbers. This library finally helps you achieve that!

```
This has been one hell of a day! The Instagram shortcode system has now
been reverse engineered. They encode the IDs using a special alphabet in
chunks of 6 bits per character. I've written a blazing-fast implementation
in pure PHP without requiring any separate math libraries.

The biggest problem was handling the fact that Instagram's media IDs are
much bigger than what fits into a CPU integer, which meant that I had to
basically write a miniature version of a "BigInteger" library from scratch.

Phew! Hope you enjoy all of this hard work! ;-)
```

**Comment by joseflorido** [comment]
Hi!

Thanks for this. Is your class producing 'better' results than this function https://github.com/postaddictme/instagram-php-scraper/blob/8d474a9227984d5437c18804e75d9d46f4d729db/src/InstagramScraper/Model/Media.php#L160  ?

I have been using this one without problems before.
Thanks!

**Comment by SteveJobzniak** [comment]
@joseflorido Yes that class is idiotic and won't work for all IDs. Written by someone who clearly doesn't know that Instagram's IDs can be larger than what fits in a CPU integer. Good luck using that one with very large IDs.

Example ID:
1038059720608660215

Largest value that class can handle on a 32-bit machine:
2147483647

So good luck with that, as I said. ;-)

**Comment by joseflorido** [comment]
Awesome, thanks again. I will replace it with your class :)

**Comment by SteveJobzniak** [comment]
Oh and we do have plenty of 32-bit users. Like @zuexo: https://github.com/mgp25/Instagram-API/issues/996

;-)

**Comment by SteveJobzniak** [comment]
2.x is being released as we speak. Closing this thread. It will be covered in the release thread.
